server/controllers/EventController.js

```diff
@@ -16,6 +15,64 @@ const getEvents = async (req, res, next) => {
 
   return res.json(events)
 }
+/*
+* Returns an array of JSON objects of a given length (10 if undefined), filled firs by all events that still need submitting, with the rest
+* of the array being filled with the most recent events ordered by date.
+* */
+const getEventDashboardList = async (req, res, next) => {
+
+  // Check if an array length number has been defined and store that value, otherwise default the value to 10
+  let listLength = (req.query.listLength !== undefined) ? req.query.listLength : 10
+
+  // Find & store all events from the collection
+  let allEvents = await getEventsFromDb()
+
+  // Find & store all events that still need submitting
+  let noRecordEvents = allEvents.filter((event) => {
+    return !event.hasAttendanceRecords
+  })
+
+  // Find & store all events that have allready been submitted
+  let recordEvents = allEvents.filter((event) => {
+    return event.hasAttendanceRecords
+  })
+
+  let sortedRecordEvents = recordEvents.sort((a, b) => {
+    const dateA = new Date(a.date)
+    const dateB = new Date(b.date)
+    return (dateA > dateB) ? -1 : 1
+  })
+
+  // Find how many recent events will fit into the final array, and reduce the array to only contain those objects
+  if (noRecordEvents.length >= listLength) {
+    recordEvents = []
+  } else if (recordEvents.length > listLength - noRecordEvents.length) {
+    recordEvents = recordEvents.slice(0, listLength - noRecordEvents.length)
+  }
+
+  let currentEvent;
+
+  let combinedEvents = [...recordEvents, ...noRecordEvents]
+
+  if (req.query.eventId !== null) {
+
+
+    // Using (==) operator here, as even._id and req.query.eventId are different types, and that is not important for the comparison.
+    const doesExist = combinedEvents.some(event => event._id == req.query.eventId)
+    if (!doesExist) {
+      currentEvent = allEvents.filter(event => event._id == req.query.eventId)
+      combinedEvents = [...combinedEvents, ...currentEvent]
+    }
+  }
+
+  let events = combinedEvents.sort((a, b) => {
+    const dateA = new Date(a.date)
+    const dateB = new Date(b.date)
+    return (dateA > dateB) ? -1 : 1
+  })
+
+  res.json(events)
```

Comment:
we can just do:

```
combinedEvents.sort((a, b) => {
  const dateA = new Date(a.date)
  const dateB = new Date(b.date)
  return (dateA > dateB) ? -1 : 1
})

res.json(combinedEvents)
```

as `.sort()` mutates the original array. It _also_ returns the sorted array, which is why this also works. Let's do the above so that we don't declare so many different variations to `events` as we've already got a few in this function.